azure-devops/azext_devops/dev/repos/policy.py

```diff
@@ -215,42 +215,76 @@ def update_policy_required_reviewer(policy_id,
                                                 'fa4e907d-c16b-4a4c-9dfa-4916e5d171ab',
                                                 param_name_array, param_value_array, branch_match_type)
-
     return policy_client.create_policy_configuration(configuration=configuration, project=project)
 
 
 def update_policy_merge_strategy(policy_id,
                                  repository_id=None, branch=None, branch_match_type=None,
                                  blocking=None, enabled=None,
                                  use_squash_merge=None,
+                                 allow_squash=None,
+                                 allow_rebase=None,
+                                 allow_rebase_merge=None,
+                                 allow_no_fast_forward=None,
                                  organization=None, project=None, detect=None):
     """Update merge strategy policy
     """
     organization, project = resolve_instance_and_project(
         detect=detect, organization=organization, project=project)
     policy_client = get_policy_client(organization)
     current_policy = policy_client.get_policy_configuration(project=project, configuration_id=policy_id)
-    param_name_array = ['useSquashMerge']
-
     current_setting = current_policy.settings
     current_scope = current_policy.settings['scope'][0]
+    if use_squash_merge is None:
+        param_name_array = ['allowSquash', 'allowRebase', 'allowRebaseMerge', 'allowNoFastForward']
+        param_value_array = [
+            allow_squash if allow_squash is not None else current_setting.get('allowSquash',
+                                                                            current_setting.get('useSquashMerge', None)),
+            allow_rebase if allow_rebase is not None else current_setting.get('allowRebase', None),
+            allow_rebase_merge if allow_rebase_merge is not None else current_setting.get('allowRebaseMerge', None),
+            allow_no_fast_forward if allow_no_fast_forward is not None else current_setting.get('allowNoFastForward',
+                                                                                                None)
+        ]
 
-    param_value_array = [
-        use_squash_merge if use_squash_merge is not None else current_setting.get('useSquashMerge', None)
-    ]
+        # We cannot send setting as None in the API
+        for i, value in enumerate(param_value_array):
+            if value is None:
+                param_value_array[i] = False
+        # API does not fail but the update is rejected if the last setting is being set to false.
+        # So this check prevents it from client side.
+        if not [i for i, value in enumerate(param_value_array) if value]:
+            raise CLIError("Atleast one merge type must be enabled.")
+    else:
+        param_name_array = ['useSquashMerge']
+        param_value_array = [use_squash_merge]
```

Comment:
use_squash_merge if use_squash_merge is not None else current_setting.get('useSquashMerge', None) ?

Reply:
Good catch!